bluemira/builders/coil_supports.py

```diff
@@ -525,6 +533,267 @@ def build_xyz(
+    def f_L_to_wire(wire, x_norm):  # noqa: N802
+        """
+        Convert a pair of normalised L values to a wire
+        """
+        p1 = wire.value_at(x_norm[0])
+        p2 = wire.value_at(x_norm[1])
+        return make_polygon([p1, p2])
+
+    @staticmethod
+    def f_L_to_xz(wire, value):  # noqa: N802
+        """
+        Convert a normalised L value to an x, z pair.
+        """
+        point = wire.value_at(value)
+        return np.array([point[0], point[2]])
+
+    @staticmethod
+    def _f_objective_ois(x_norm, wire):
+        p1 = StraightOISOptimisationProblem.f_L_to_xz(wire, x_norm[0])
+        p2 = StraightOISOptimisationProblem.f_L_to_xz(wire, x_norm[1])
+        return -np.hypot(*(p2 - p1))
+
+    @staticmethod
+    def f_objective_ois(x_norm, grad, wire):
+        """
+        Maximise the length of a straight line
+        """
+        value = StraightOISOptimisationProblem._f_objective_ois(x_norm, wire)
+        if grad.size > 0:
+            grad[:] = approx_derivative(
+                StraightOISOptimisationProblem._f_objective_ois,
+                x_norm,
+                f0=value,
+                bounds=[(0, 0), (1, 1)],
+                args=(wire,),
+            )
+
+        return value
+
+    @staticmethod
+    def _f_constraint_ois(x_norm, wire, koz_points):
+        straight_line = StraightOISOptimisationProblem.f_L_to_wire(wire, x_norm)
+        straight_points = straight_line.discretize(ndiscr=100).xz.T
+        return signed_distance_2D_polygon(straight_points, koz_points)
+
+    @staticmethod
+    def f_constraint_ois(constraint, x_norm, grad, wire, koz_points):
+        """
+        Constraint the OIS to be outside of a keep out zone
+        """
+        constraint[:] = StraightOISOptimisationProblem._f_constraint_ois(
+            x_norm, wire, koz_points
+        )
+        if grad.size > 0:
+            grad[:] = approx_derivative(
+                StraightOISOptimisationProblem._f_constraint_ois,
+                x_norm,
+                f0=constraint,
+                bounds=[(0, 0), (1, 1)],
+                args=(wire, koz_points),
+            )
+        return constraint
+
+    @staticmethod
+    def f_constraint_x(constraint, x_norm, grad):
+        """
+        Constrain the second normalised value to be always greater than the first.
+        """
+        constraint[0] = x_norm[0] - x_norm[1]
+
+        if grad.size > 0:
+            grad[:, 0] = 1.0
+            grad[:, 1] = -1.0
+
+        return constraint
+
+
+@dataclass
+class StraightOISDesignerParams(ParameterFrame):
+    """
+    Parameters for the StraightOISDesigner
+    """
+
+    tk_ois: Parameter[float]
+    g_ois_tf_edge: Parameter[float]
+    min_OIS_length: Parameter[float]
+
+
+class StraightOISDesigner(Designer[List[BluemiraWire]]):
+    """
+    Design a set of straight length outer inter-coil structures.
+
+    Parameters
+    ----------
+    params
+    build_config
+    tf_coil_xz_face
+    keep_out_zones
```

Comment:
This is autofilled by sphinx in the docs so I would say isnt needed if there is no extra description
```suggestion
```